.github/workflows/tests.yml

```diff
@@ -0,0 +1,62 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python package
+
+on: [push, pull_request]
+  # push:
+  #   branches: [ master ]
+  # pull_request:
+  #   branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install .[sphinx,code_style,testing]
+    - name: Pre-commit checks
+      run: |
+        pre-commit run --all-files || ( git status --short ; git diff ; exit 1 )
+    # cd tests/test_commonmark && ./spec.sh
+    - name: Test with pytest
+      run: |
+        pip install pytest
+        pytest --cov=myst_parser --cov-report=
+    - name: Upload to coveralls
+      run: |
+        pip install coveralls
+        coveralls
+      env:
+          COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_KEY }}
+
+  publish:
+
+    name: Publish to PyPi
+    needs: build
+    if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
```

Comment:
@choldgraf, you can give conditionals to each job. In this case the 'publish' is triggered conditional on the 'build' job passing, and if the event was triggered by creating a release tag on the repo

Reply:
shit that's cool! I'm really liking github-actions (I know I know, you're not allowed to make fun of me every time I praise a microsoft product)

Reply:
😆 
yeh well in terms of document building, its certainly something we could consider providing our own workflow/action for; something like https://github.com/ammaraskar/sphinx-action (using a Docker environment), but also making use of caching and artefacts: https://help.github.com/en/actions/configuring-and-managing-workflows/caching-and-storing-workflow-data

Reply:
oh I am one step ahead of you on that one ;-)  even have a friend that works on Microsoft helping out!

https://github.com/ExecutableBookProject/cli/issues/44